src/node/internal/internal_zlib.ts

```diff
@@ -20,6 +20,250 @@ const {
   CONST_UNZIP,
 } = zlibUtil;
 
+export function crc32(
+  data: ArrayBufferView | string,
+  value: number = 0
+): number {
+  validateUint32(value, 'value');
```

Comment:
let's throw on overly large inputs. Anything larger than 128 * 1024 should just automatically be rejected... even that is being very generous.